### ISSUE TITLE
Support flatpak installation on upstrema flathub

### ISSUE THREAD

**Issue by yodatak** [Sigil-Ebook/Sigil#521]
Hi i could be cool to have the way to install sigil with flatpak , for know the version in fedora 31 by exemple is 0.9.14 
There is a community effort to make a flatpak in flathub 

https://github.com/flathub/flathub/pull/1211

It cloud be interesting to support it upstream

**Comment by dougmassay** [comment]
We neither provide nor endorse Sigil flatpak builds. It's fine if others want to take that up, but we have neither the time to participate, nor the interest.

**Comment by eli-schwartz** [comment]
I agree, the flathub maintainers are the best people to provide support for thirdparty flatpaks, the Sigil developers only need to concern themselves with providing build documentation and being available for asking questions. ;)

@yodatak if Fedora hasn't updated to current releases of Sigil, it is best to ask the Fedora maintainers about it. There's a Fedora tracking bug for the new version here: https://bugzilla.redhat.com/show_bug.cgi?id=1724109

**Comment by catsout** [comment]
@dougmassay
https://github.com/flathub/flathub/pull/2819

**Comment by kevinhendricks** [comment]
Interesting.  Nicely done!

FYI:

- a new Sigil release is coming (1.9) soon and it uses pcre2 and python 3.9.9

- our Windows and macOS builds add custom bug fix patches to our embedded Qt builds.  We support patches for qt 5.12.9 and Qt 6.2.2 at the moment.

- our Windows and macOS support our own custom hunspell dictionaries

- filesystem access for photos, video, audio would also be useful as these can be sources for files when creating epubs

**Comment by catsout** [comment]
> our Windows and macOS builds add custom bug fix patches to our embedded Qt builds. We support patches for qt 5.12.9 and Qt 6.2.2 at the moment.
Sigil-1.9 will also build with Qt6.2.x if that helps in any way.

There is qt 6.x runtime in flathub, so it's ok to update to 1.9. 
So is it fine to use qt 6.x without patches?  

>  our Windows and macOS support our own custom hunspell dictionaries

I think as long as we use the bundled hunspell in sigil, linux will be same as windows.

**Comment by kevinhendricks** [comment]
In Sigil/docs we have 2 Qt patches needed for non-macOS programs.  Both are to fix bugs long ago reported to Qt that were never addressed by them.  The first prevents a segfault when the user closes multiple tabs or changes active tabs very quickly.  The second fixes a bug in Chrome's/WebEngine dealing with non-spec compliant live editing behaviour when using h6 elements.  This second patch is more useful for Sigil's companion program PageEdit.

**Comment by catsout** [comment]
I don't have time to maintain the qt and qtwebengine.  
Let's keep the same as other linux distribution.

**Comment by nedrichards** [comment]
It's worth noting that the flathub KDE runtime uses the Qt maintained by the KDE project which also contains downstream patches and fixes, especially for 5.15. https://invent.kde.org/qt/qt/qtbase/-/merge_requests etc. that's a good concentrator of fixes for LTS branches where the Qt company is not providing releases if you have a patch you want to share.

**Comment by kevinhendricks** [comment]
@nedrichards 
It is my understanding, that 5.15 kde tree will only accept changes made officially in the upstream Qt tree.

Our patches can be found in Sigil/docs here and both have officially filed bugs with Qt with patches to fix them.

```
# apply mouse press and flick on qtabbar tab fix to prevent segfault
# See https://bugreports.qt.io/browse/QTBUG-74478
patch -p0 < ./qt512.8_avoid_qtabbar_segfault.patch

# then apply an obvious fix for insertParagraph execcommand and h6 (see w3c spec)
# See https://bugreports.qt.io/browse/QTBUG-79778
patch -p0 < ./qt512.8_fix_h6_insertParagraph.patch
```

We have been shipping our own Qt with those fixes and even more macOS specific ones, literally for years.  Those two are simple obvious patches but getting Qt to fix anything is next to impossible.  I have been reporting and supplying patches for Qt bugs for years but they really do not give a rat's ass about stability, as seen by their recent decision to withhold LTS releases.  I thought Qt was obligated to release their LTS versions as opensource after one year but have seen nothing later from 5.15 LTS.

Versions of those patches for Qt 6.2.x are in Sigil docs as well.  Good luck trying to get someone to include them.
